### PR TITLE
Change layout of environment building directory

### PR DESCRIPTION
**Pull request by jherbel**

### CODE REVIEW

File: examples/termination/main.rs
[the remark this answers is not in the document]
Comment (SoloJacobs) [reply examples/termination/main.rs:1]: `handling of this runtime directory more similar to the handling of the runtime directory...` reads very weirdly.

File: src/environment.rs [hunk 1]
```diff
@@ -12,7 +12,10 @@ pub enum ResultCode {
 #[derive(Clone, Debug, PartialEq)]
 pub enum Environment {
     System(SystemEnvironment),
-    Rcc(RCCEnvironment),
+    Rcc(
+        // see https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant
```
Comment (SoloJacobs) [inline src/environment.rs:16]: It says always measure the changes this lint suggests ? From what I can tell boxing is a premature optimization here, I would just suppress it.

File: src/bin/scheduler/setup/general.rs [hunk 2]
```diff
@@ -183,6 +187,61 @@ fn setup_rcc_working_directories(
     )
 }
 
+fn setup_environment_building_directories(
```
Comment (SoloJacobs) [inline src/bin/scheduler/setup/general.rs:190]: Why is there so much extra code here? Is this change adding functionality?

File: examples/termination/main.rs
[the remark this answers is not in the document]
Comment (jherbel) [reply examples/termination/main.rs:1]: Done

File: src/environment.rs (hunk 1 above)
Comment (jherbel) [reply src/environment.rs]: Done

File: src/bin/scheduler/setup/general.rs (hunk 2 above)
Comment (jherbel) [reply src/bin/scheduler/setup/general.rs:190]: Not really, but I found it hard to consolidate this with the already existing functionality. I would suggest that we first make the necessary changes and then check if and how we can reduce code duplication. Ok?

**Review by SoloJacobs — APPROVED** [review]
Nice!

File: src/bin/scheduler/setup/general.rs (hunk 2 above)
Comment (SoloJacobs) [reply src/bin/scheduler/setup/general.rs:190]: Yes, that does not bother me. It's not really about the code duplication, but I think we might need better interfaces for adding functionality.

**Pull request merged**